### ISSUE TITLE
How to handle `source_hidden` and `outputs_hidden`

### ISSUE THREAD

**Issue by ellisonbg** [jupyterlab/jupyterlab#2740]
Opened for discussion about these new cell metadata fields that was brought up at the weekly team meeting.

**Comment by ellisonbg** [comment]
The initial thought is as follows:

* There would be at least two abstractions/layers for persisting this information. This is needed because sometimes this information does need to be saved in the document, and other times it doesn't. Also, users probably woudn't want this information RTC synced.
* The new metadata in the nbformat is only for the part of it that is persisted to the notebook document.
* Our input/output collapser UI/UX in JupyterLab would not modify the saved document state.
* We can and probably should store the collapsed state of our UI/UX in a location that is not real-time synced. Local store maybe? Maybe the new modeldb APIs can have a flag that can be set on attributes that shouldn't be RTC'd?
* We would have a specific command (and UI) for persisting/loading a user's current UI/UX collapsed state into the notebook metadata.

In summary: we will need non real-time state of the input/output collapsed state, that is also not persisted in the notebook document unless a user takes a specific action requesting it to be stored there.

@sccolbert If I understood what you were saying, I think this addresses your concerns?

**Comment by ellisonbg** [comment]
Moving this to beta to make sure we are handling this metadata in a reasonable manner.

**Comment by ellisonbg** [comment]
@blink1073 let's chat about this in the am to see if we want to do anything for the beta...

**Comment by ellisonbg** [comment]
We will also run into these types of questions more generally when we merge the datastore work. Deferring until then...
